### PR TITLE
[update to #992] Fix and refactor MoveObjectReward 

### PR DESCRIPTION
**Pull request by vincentpierre**
## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by ASzot — APPROVED** [review]
I've tested this and it works as expected for me.

File: habitat-lab/habitat/tasks/rearrange/multi_task/composite_sensors.py [hunk 1]
```diff
@@ -108,52 +125,55 @@ def update_metric(self, *args, episode, task, observations, **kwargs):
...
-                self._cur_rearrange_step, num_targs - 1
-            )
+        # PLACE REWARD: Reward for placing the object correcly (within success dist)
+        # We also udate the target object for the next stage.
 
-        self._metric += self._config.dist_reward * dist_diff
-        self._prev_measures = (to_obj, to_goal)
+        place_success = obj_to_goal_dist < self._config.success_dist
+        if place_success and not is_holding_obj:
+            self._metric += self._config.single_rearrange_reward
```
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/rearrange/multi_task/composite_sensors.py:164]: @SergioArnaud Is it possible to double count the place reward here if the task does not end after placing? I would add a `if self._cur_rearrange_stage < self.num_targets` to make sure of it.

Comment (SergioArnaud) [reply habitat-lab/habitat/tasks/rearrange/multi_task/composite_sensors.py:164]: @vincentpierre I think that case is considered in [this](https://github.com/vincentpierre/habitat-lab/blob/900629559acec863a70782b7372e21b77e5d3870/habitat-lab/habitat/tasks/rearrange/multi_task/composite_sensors.py#L131-L133) condition. If all the objects are placed in the right place but for some reason we haven't succeded in the task (for example the agent hasn't called terminate) we give zero reward.

Comment (vincentpierre) [reply habitat-lab/habitat/tasks/rearrange/multi_task/composite_sensors.py:164]: Oh, nice. sorry I missed it.

**Pull request merged**